file_optical.c

```diff
@@ -1670,13 +1670,60 @@ static int fbo_handle_cmd(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 	return 0;
 }
 
+static int fbo_update_cfgstring(struct tcmu_device *dev, struct tcmulib_cfg_info *cfg)
+{
+	char *oldptr;
+
+	if (!tcmu_check_state(dev, TCMU_CAPACITY_CHANGE_FOR_CONFIG_CHANGE)) {
+		tcmu_err("You need to include size change to be able to change dev_config\n");
+		return -EINVAL;
+	}
+	tcmu_remove_state(dev, TCMU_CAPACITY_CHANGE_FOR_CONFIG_CHANGE);	
+
+	tcmu_update_state(dev, TCMU_CONFIG_CHANGING);
+	
+	fbo_close(dev);
+
+	oldptr = cfg->data.dev_cfgstring;
+	tcmu_set_dev_cfgstring(dev, oldptr);
+
+	tcmu_err("path is %s\n", oldptr);
+
+	fbo_open(dev);
```

Comment:
Why not return an error if this fails?

Maybe at least an error message if this fails.